### ISSUE TITLE
[BUG] NPE in azure-storage-file-datalake-12.0.0-beta.10

### ISSUE THREAD

**Issue by ylemoigne** [Azure/azure-sdk-for-java#7653]
**Describe the bug**
NPE in azure-storage-file-datalake-12.0.0-beta.10

***Exception or Stack Trace***

```
Exception in thread "main" java.lang.NullPointerException
	at com.azure.storage.file.datalake.Transforms.toPathItem(Transforms.java:257)
```

**To Reproduce**
List recursivly a datalake.

To be more specific, the `path.getContentLength()` return a `Long` coerced to `long` in `PathItem` constructor. When the `path.getContentLength()` is null (which happen) the coersion throw NPE.

**Setup (please complete the following information):**
Bug in version 12.0.0-beta.8 and 12.0.0-beta.10 at least

**Additional context**

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Bug Description Added
- [x] Repro Steps Added
- [x] Setup information Added

**Comment by gapra-msft** [comment]
Hi @ylemoigne 

Thank you for reporting this issue. We will schedule some time to get a fix in for this soon.

**Comment by ylemoigne** [comment]
Thank you :)

**Comment by xARCHIMEDESx** [comment]
@ylemoigne thanks for reporting the bug. I strugled with it all yesterday day thinking I am doing something wrong. Then decided to check open issues here)
@gapra-msft thanks for quick fixing. When 12.0.0-beta.11 can be expected to be released?  A am developing an Azure Function app which accesses to ADLSgen2 and this bug is a blocker for my task. Thanks for an answer.

**Comment by gapra-msft** [comment]
Hi @xARCHIMEDESx You can expect the next release in the next couple weeks or so. I will update this thread when we release.

**Comment by gapra-msft** [comment]
This fix was released as part of 12.0.0-beta.12
